### DISCUSSION TITLE
on_load阶段如何获取编译器路径

### DISCUSSION THREAD

**Discussion by PeakRacing** [xmake-io/xmake#4578] [Q&A] [answered]
on_load阶段如何获取编译器路径呢？我之前在before_link阶段toolchains = target:toolchains()[1]:bindir()获取，但是on_load阶段这还没生效

**Comment by waruqi** [comment, marked as answer]
target:tool("cc") ，但不建议 on_load ，至少也要在  after_load 或者 on_config 中获取

**Reply by PeakRacing** [reply]
target:add "defines" “linkgroups” 这些可以在 after_load 或者 on_config 中生效么？

**Reply by waruqi** [reply]
能

**Reply by PeakRacing** [reply]
感谢
